charging_stations_pipelines/pipelines/osm/osm_mapper.py

```diff
@@ -28,35 +39,176 @@ def map_station_osm(entry: Dict, country_code: str):
     return new_station
 
 
-def map_address_osm(entry, station_id):
-    if "tags" in entry:
-        tags = entry["tags"]
-        if (
-                "addr:city" in tags
-                and "addr:country" in tags
-                and "addr:housenumber" in tags
-                and "addr:postcode" in tags
-                and "addr:street" in tags
-        ):
-            city = tags["addr:city"]
-            country = tags["addr:country"]
-            housenumber = tags["addr:housenumber"]
-            postcode = tags["addr:postcode"]
-            street = tags["addr:street"]
-            map_address = Address()
-            map_address.state = None
-            map_address.station_id = station_id
-            map_address.street = street + " " + housenumber
-            map_address.town = city
-            map_address.postcode = postcode
-            map_address.district = None
-            map_address.country = country
-            return map_address
-    return None
-
-
-def map_charging_osm(station_id):
+def map_address_osm(entry: Dict, station_id: Optional[int]) -> Optional[Address]:
+    """This method maps the address information from the OpenStreetMap (OSM) entry to an Address object.
+
+    :param entry: A dictionary representing the address entry
+    :param station_id: An optional integer representing the station ID
+    :return: An optional Address object
+    """
+    tags = entry.get('tags')
+    if not tags:
+        return None
+    address_keys = [  # Final[list[LiteralString]]
+        "addr:city",
+        "addr:country",
+        "addr:housenumber",
+        "addr:postcode",
+        "addr:street"]
+    if all(key in tags for key in address_keys):
+        city = tags["addr:city"]
+        country = tags["addr:country"]
+        housenumber = tags["addr:housenumber"]
+        postcode = tags["addr:postcode"]
+        street = tags["addr:street"]
+
+        map_address = Address()
+        # TODO add state_old to Address
+        # map_address.state_old = None
+        map_address.station_id = station_id
+        map_address.street = street + " " + housenumber
+        map_address.town = city
+        # TODO add postcode to Address
+        # map_address.postcode = postcode
```

Comment:
Why is this commented? I assume there is an issue with this field.

Reply:
I clarified this with @mjmader... this field should be used (again).